AudioChannel.js

```diff
@@ -0,0 +1,126 @@
+//▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
+/** Audio Channel class.
+ *
+ * @author  Cedric Stoquer
+ *
+ *
+ * @param {string} id - channel name id
+ */
+function AudioChannel(id) {
+	this.id        = id;
+	this.volume    = 1.0;
+	this.muted     = true;
+	this.loopSound = null;
+	this.loopId    = null;
+	this.loopVol   = 0.0;
+	this.nextLoop  = null;
+}
+
+
+//▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
+AudioChannel.prototype.setVolume = function (volume, muted) {
+	var wasChannelMuted = this.muted;
+	this.muted  = volume === 0 || muted || false;
+	if (volume !== undefined && volume !== null) {
```

Comment:
Up to you, but would it make more sense to check for `typeof volume === 'number'`?